python_common_logger/src/flask/middleware.py

```diff
@@ -0,0 +1,37 @@
+from uuid import uuid4
+from ..context.context_handler import update_execution_context
+from ..context.execution_context import ExecutionContext
+
+from ..constants.request import RequestHeaderKeys
+from ..constants.context import ExecutionContextType
+
+
+class FlaskContextMiddleware():
```

Comment:
Also, isn't it better to use the WSGI middleware rather than before_request? Seems cleaner as it is directly added to the app rather than adding a before request:

https://ohadp.com/adding-a-simple-middleware-to-your-flask-application-in-1-minutes-89782de379a1
https://blog.caoyu.info/middleware-in-flask.html

Reply:
Implemented adjustments according to WSGI middleware
now the application side implementation will be like:
`application.wsgi_app = FlaskContextMiddleware(application.wsgi_app)`
This new approach supersedes the utilization of the before_request hook